### PR TITLE
k8s-reporter: Stop dumping unreachable container's info

### LINKED ISSUE #7799: Reporter provides unhelpful information / noise if virt-launcher is gone
**What happened**:

When virt-launcher is gone by the time the reporter is executed, we log several lines of unhelpful logs.

The following was logged after failed `[sig-compute]User Access [rfe_id:2919][crit:high][vendor:cnv-qe@redhat.com][level:component] With regular OpenShift user should succeed with admin rights for the project should verify permissions on resources are correct for subresources [test_id:2921]given a vmi (guestosinfo)` test due to `ERR: Error from server (AlreadyExists): users.user.openshift.io "testuser-kubevirt-test-default3" already exists`:

```
failed to execute command [ip address] on virt-launcher-testvmi-k7pqp-qkkhv, stdout: , stderr: , error: Internal error occurred: error executing command in container: container is not created or running
failed to execute command [ip link] on virt-launcher-testvmi-k7pqp-qkkhv, stdout: , stderr: , error: unable to upgrade connection: container not found ("compute")
failed to execute command [ip route show table all] on virt-launcher-testvmi-k7pqp-qkkhv, stdout: , stderr: , error: unable to upgrade connection: container not found ("compute")
failed to execute command [ip neigh show] on virt-launcher-testvmi-k7pqp-qkkhv, stdout: , stderr: , error: unable to upgrade connection: container not found ("compute")
failed to execute command [bridge -j vlan show] on virt-launcher-testvmi-k7pqp-qkkhv, stdout: , stderr: , error: unable to upgrade connection: container not found ("compute")
failed to execute command [bridge fdb] on virt-launcher-testvmi-k7pqp-qkkhv, stdout: , stderr: , error: unable to upgrade connection: container not found ("compute")
failed to execute command [lspci] on virt-launcher-testvmi-k7pqp-qkkhv, stdout: , stderr: , error: unable to upgrade connection: container not found ("compute")
failed to execute command [env] on virt-launcher-testvmi-k7pqp-qkkhv, stdout: , stderr: , error: unable to upgrade connection: container not found ("compute")
failed to execute command [[ -e /dev/vfio ] && ls -lsh -Z -St /dev/vfio] on virt-launcher-testvmi-k7pqp-qkkhv, stdout: , stderr: , error: unable to upgrade connection: container not found ("compute")
failed to execute command [ip address] on virt-launcher-testvmi-k7pqp-whpjs, stdout: , stderr: , error: pods "virt-launcher-testvmi-k7pqp-whpjs" not found
failed to execute command [ip link] on virt-launcher-testvmi-k7pqp-whpjs, stdout: , stderr: , error: pods "virt-launcher-testvmi-k7pqp-whpjs" not found
failed to execute command [ip route show table all] on virt-launcher-testvmi-k7pqp-whpjs, stdout: , stderr: , error: pods "virt-launcher-testvmi-k7pqp-whpjs" not found
failed to execute command [ip neigh show] on virt-launcher-testvmi-k7pqp-whpjs, stdout: , stderr: , error: pods "virt-launcher-testvmi-k7pqp-whpjs" not found
failed to execute command [bridge -j vlan show] on virt-launcher-testvmi-k7pqp-whpjs, stdout: , stderr: , error: pods "virt-launcher-testvmi-k7pqp-whpjs" not found
failed to execute command [bridge fdb] on virt-launcher-testvmi-k7pqp-whpjs, stdout: , stderr: , error: pods "virt-launcher-testvmi-k7pqp-whpjs" not found
failed to execute command [lspci] on virt-launcher-testvmi-k7pqp-whpjs, stdout: , stderr: , error: pods "virt-launcher-testvmi-k7pqp-whpjs" not found
failed to execute command [env] on virt-launcher-testvmi-k7pqp-whpjs, stdout: , stderr: , error: pods "virt-launcher-testvmi-k7pqp-whpjs" not found
failed to execute command [[ -e /dev/vfio ] && ls -lsh -Z -St /dev/vfio] on virt-launcher-testvmi-k7pqp-whpjs, stdout: , stderr: , error: pods "virt-launcher-testvmi-k7pqp-whpjs" not found
skipping pod virt-launcher-testvmi-56xtv-pmn69, phase is not Running
```

**What you expected to happen**:

We should keep the logs clean to allow faster debugging and identification of the issue. Instead of the noise above, we should just log once that the pod is not running and we cannot collect information from it.

### PR DESCRIPTION
**Pull request by oshoval**
**What this PR does / why we need it**:
In order to keep to logs cleaner and more informative,
refrain from printing the same error again again,
and also save time if the container / pod aren't
reachable (instead of running more commands).

Additionally - Fix logic of checking if the virt-handler's compute container is healthy,
not just exist.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #7799 

**Special notes for your reviewer**:

**Release note**:
```release-note
None
```

### CODE REVIEW

File: tests/reporter/kubernetes.go [hunk 1]
```diff
@@ -1104,6 +1104,10 @@ func (r *KubernetesReporter) executeContainerCommands(virtCli kubecli.KubevirtCl
 				failedExecuteCmdFmt,
 				command, target, stdout, stderr, err,
 			)
+
+			if strings.Contains(err.Error(), "not found") {
```
Comment (AlonaKaplan) [inline tests/reporter/kubernetes.go:1108]: Is there a reason not to use `errors.IsNotFound`?

Comment (oshoval) [reply tests/reporter/kubernetes.go:1108]: It isn't a `IsNotFound` type, it is created by
`fmt.Errorf("unable to upgrade connection: %s", responseError)`
the `not found` is a string that appears in both cases of container and pod not found (see the full lines please in the ref issue)

Comment (AlonaKaplan) [reply tests/reporter/kubernetes.go:1108]: What worries me is that we may catch another error with `not found` that is not one of the two this PR wants to catch.
On the other side, if the messages we want to catch will be changed (for example to "Not Found") the condition will stop working.

To solve at least the first issue let's use the full two messages.

**Comment by oshoval** [comment]
Addressed comments by checking that the pod / container are healthy,
fixed another place that checked only that the compute container exists, but didn't check it is healthy

**Comment by oshoval** [comment]
Rebased

File: tests/reporter/kubernetes.go (hunk 1 above)
Comment (oshoval) [reply tests/reporter/kubernetes.go]: [Changes](https://github.com/kubevirt/kubevirt/pull/7815#issuecomment-1142236264)

**Comment by xpivarc** [comment]
Is this ready?

**Comment by oshoval** [comment]
> Is this ready?

Basically yes, i want to do few more tests, once it will be ready i will undraft

**Comment by oshoval** [comment]
Update inner `pod` variable to not override the outer one, so if there is a Get error, the outer won't be affected,
as it is still being used

File: tests/reporter/kubernetes.go [hunk 2]
```diff
@@ -1104,6 +1106,11 @@ func (r *KubernetesReporter) executeContainerCommands(virtCli kubecli.KubevirtCl
 				failedExecuteCmdFmt,
 				command, target, stdout, stderr, err,
 			)
+
+			pod, err := virtCli.CoreV1().Pods(pod.ObjectMeta.Namespace).Get(context.Background(), pod.ObjectMeta.Name, metav1.GetOptions{})
+			if errors.IsNotFound(err) || (err == nil && (pod.Status.Phase != "Running" || !isContainerReady(pod.Status.ContainerStatuses, container))) {
```
Comment (AlonaKaplan) [inline tests/reporter/kubernetes.go:1111]: Did you tell me the error is not of kind `errors.IsNotFound`?

Why not just - `if err != nil && (pod.Status.Phase != "Running" || !isContainerReady(pod.Status.ContainerStatuses, container)))`?

Comment (oshoval) [reply tests/reporter/kubernetes.go:1111]: > Did you tell me the error is not of kind errors.IsNotFound?
it is not the same API call, now we are doing a client.Get, before that the error was the Execute on pod

about the 2nd question
if err != nil, you can't look on the result even, it is undefined / potential not exists at all, anyhow the result is not valid as the Get failed (hence err != nil)

**Comment by oshoval** [comment]
Thanks Alona

@qinqon can you please review again ?

**Pull request merged**